lib/puppet/util/checksums.rb

```diff
@@ -198,6 +195,29 @@ def none(content)
     ""
   end
 
+  class DigestLite
+    def initialize(digest, lite = false)
+      @digest = digest
+      @lite = lite
+      @bytes = 0
+    end
+
+    # Provide an interface for digests. If lite, only digest the first 512 bytes
+    def <<(str)
+      if @lite
+        if (str.length + @bytes) < 512
+          @digest << str
+          @bytes += str.length
+        elsif @bytes < 512
+          @digest << str[0, 512-@bytes]
+          @bytes = 512
+        end
```

Comment:
Something doesn't look right with the `elsif` case. If we're doing md5lite, I think there would be an else case that ignored `str` once `@bytes == 512`. Perhaps something like this?

```
if @bytes < 512
  if (str.length + @bytes) <= 512
    buf = str
  else
    buf = str[0, 512 - @bytes]
  end

  @digest << buf
  @bytes += buf.length
end
```

In the first `if` case, if the `str.length` + current position is less than or *equal* to the max length, then we can process the entire `str`. Else we'd be exceeding the max length, and would need to substring (copy) the buffer.

Alternatively, we could always substring (as it's only the lite case):

```
if @bytes < 512
  buf = str[0, 512 - @bytes]
  @digest << buf
  @bytes += buf.length
end
```